### ISSUE TITLE
The declaration file is missing the static tag

### ISSUE THREAD

**Issue by kongLau** [wessberg/rollup-plugin-ts#62]
Hi,
The dts file emit with plugin is missing the static tag

> Pool.ts
`class PoolImpl<T extends { dispose?(): void, onAlloc?(): void, onRelease?(): void }> {`
`    constructor(cls: new() => T) {}`
`    /* @internal */ public alloc(): T {}`
`    /* @internal */ public release(item: T): void {}`
`    /* @internal */ public resize(time: number): void {}`
`    /* @internal */ public dispose(): void {}`
`}`
`export class Pool {`
`    public static alloc<T>(cls: new () => T): T {}`
`    public static release(item: any): null {}`
`    public static releaseList(list: any[]): void {}`
`}`

> Pool.d.ts emit with rollup-plugin-ts@v1.2.0
`declare class Pool {`
`    alloc<T>(cls: new () => T): T;`
`    release(item: any): null;`
`    releaseList(list: any[]): void;`
`}`

> Pool.d.ts emit with tsc
`export declare class Pool {`
`    static alloc<T>(cls: new () => T): T;`
`    static release(item: any): null;`
`    static releaseList(list: any[]): void;`
`}`

**Comment by wessberg** [comment]
Hey there. I can't reproduce this problem, but I do see a few things. First, your code is not valid TypeScript and will result in a TypeScript error: `A function whose declared type is neither 'void' nor 'any' must return a value.'`.

However, the other thing is that the output you got from `rollup-plugin-ts` looks a bit weird. An ExportDeclaration is missing.

Which Node version are you running, and which OS are you using?

I've tried to reproduce this issue on Node 13 on Mac OS 10.15.3. Here's what I got:

The following code:

```typescript
class PoolImpl<T extends { dispose?(): void, onAlloc?(): void, onRelease?(): void }> {
    constructor(cls: new() => T) {}
    /* @internal */ public alloc(): T {}
    /* @internal */ public release(item: T): void {}
    /* @internal */ public resize(time: number): void {}
    /* @internal */ public dispose(): void {}
}
export class Pool {
    public static alloc<T>(cls: new () => T): T {}
    public static release(item: any): null {}
    public static releaseList(list: any[]): void {}
}
```

Will produce the following declaration file:

```typescript
declare class Pool {
    static alloc<T>(cls: new () => T): T;
    static release(item: any): null;
    static releaseList(list: any[]): void;
}
export {Pool};
```

**Comment by kongLau** [comment]
Hi, I post the sample at [here](https://github.com/kongLau/poolsample).

**Comment by kongLau** [comment]
Node v8.10.0
Windows 10 1909
TypeScript v3.7.4

**Comment by marijnh** [comment]
I'm seeing the same. It appears a class has to be re-exported for it to occur. I.e. an `index.ts` that just does `export {Foo} from "./foo"`, and then a class `Foo` in `foo.ts` that has a static method, for example `export class Foo { static m() { return 1 } }`. You get a declaration file like this:

```
declare class Foo {
    m(): number;
}
export { Foo };
```

Which I assume is cause by dropping the static flag when copying the declaration around somewhere.

**Comment by wessberg** [comment]
Related to [this issue I just opened at the TypeScript repo](https://github.com/microsoft/TypeScript/issues/35959). Will implement a workaround.

**Comment by wessberg** [comment]
This has been fixed in [v1.2.1](https://github.com/wessberg/rollup-plugin-ts/releases/tag/v1.2.1)
